pulpcore/pulpcore/app/viewsets/custom_filters.py

```diff
@@ -3,10 +3,37 @@
 """
 from django_filters import filters
 
+from pulpcore.app.viewsets import GenericNamedModelViewSet
+
 
 class CharInFilter(filters.BaseInFilter, filters.CharFilter):
     """
     Enables the user to filter a field by comma separated strings, allowing them to retrieve more
     than one object in a single query.
     """
     pass
+
+
+class HyperlinkRelatedFilter(filters.Filter):
+    """
+    Enables a user to filter by a foreign key using that FK's href
+    """
+
+    def filter(self, qs, value):
+        """
+        Args:
+            qs (django.db.models.query.QuerySet): The Queryset to filter
+            value (string): href representing an instance of foreign key
+
+        Returns:
+            Queryset filtered by the foreign key instance
+        """
+
+        if not value:
+            return qs
+
+        model_class = self.model._meta.get_field(self.name).rel.to
+        instance = GenericNamedModelViewSet.get_resource(value, model_class)
+
+        key = "{}__pk".format(self.name)
+        return qs.filter(**{key: instance.pk})
```

Comment:
I think getting the instance from the DB just to fetch its pk to construct a query is an overkill and might introduce performance issues; better might be to duplicate the URL resolving code from `GenericNamedModelViewSet.get_resource()`:
```python
match = resolve(urlparse(value).path)
key = "{}__pk".format(self.name)
return qs.filter(**{key: match.kwargs['pk']})
```

Even better might be to draw that code from `GenericNamedModelViewSet.get_resource()` into a new static method e.g `GenericNamedModelViewSet.get_view_by_url()` and use that instead